hudi-common/src/main/java/org/apache/hudi/avro/MercifulJsonConverter.java

```diff
@@ -187,196 +178,774 @@ private static Object convertJsonToAvroField(Object value, String name, Schema s
       throw new HoodieJsonToAvroConversionException(null, name, schema, shouldSanitize, invalidCharMask);
     }
 
-    JsonToAvroFieldProcessor processor = FIELD_TYPE_PROCESSORS.get(schema.getType());
-    if (null != processor) {
+    return JsonToAvroFieldProcessorUtil.convertToAvro(value, name, schema, shouldSanitize, invalidCharMask);
+  }
+
+  private static class JsonToAvroFieldProcessorUtil {
```

Comment:
Are you porting the logic from another library?  If so, could you reference the source?

Reply:
What's the reason of porting the code?

Reply:
All the code is from the same code change that are pending rolling out to hudi internal

hudi internal PR https://github.com/onehouseinc/hudi-internal/pull/697/commits/115dcef8d681bdfd4cc1f343619669600e912623

All code is written by myself, except some part is code refactoring of some existing class.